rules/prefer-type-error.js

```diff
@@ -0,0 +1,127 @@
+'use strict';
+
+const tcIdentifiers = new Set([
+	'isArguments',
+	'isArray',
+	'isArrayBuffer',
+	'isArrayLike',
+	'isArrayLikeObject',
+	'isBoolean',
+	'isBuffer',
+	'isDate',
+	'isElement',
+	'isEmptyObject',
+	'isError',
+	'isFinite',
```

Comment:
I don't mind keeping it this way. `number` in JS is very large, and contains multiple number sub-types: positive numbers, negative numbers, floating numbers, infinites and not numbers (NaN. Actually, maybe this is the definition of an imaginary number :thinking: :open_mouth:).

I could very well imagine a function saying: "I'm expecting a finite number. You gave me an infinite number, which is not the correct type". I'm :+1: with keeping this.